src/packages/core/components/input-upload-field/input-upload-field.element.ts

```diff
@@ -146,38 +208,76 @@ export class UmbInputUploadFieldElement extends FormControlMixin(UmbLitElement)
-		)}`;
+	#renderFiles() {
+		console.log('files', this._files);
+		return repeat(
+			this._files,
+			(path) => path,
+			(path) => this.#renderFile(path),
+		);
 	}
 
-	#renderFilesUploaded() {
-		if (!this._currentFiles.length) return nothing;
+	#renderFile(file: { path: string; unique: string; queueItem?: TemporaryFileQueueItem; file?: File }) {
+		// TODO: Get the mime type from the server and use that to determine the file type.
+		const type = this.#getFileTypeFromPath(file.path);
+
 		return html`
-			${repeat(
-				this._currentFiles,
-				(item) => item.unique + item.status,
-				(item) =>
-					html`<div style="position:relative;">
-						<umb-input-upload-field-file .file=${item.file as any}></umb-input-upload-field-file>
-						${item.status === 'waiting' ? html`<umb-temporary-file-badge></umb-temporary-file-badge>` : nothing}
-					</div> `,
-			)}
-		</div>`;
+			<div style="position:relative; display: flex; width: fit-content; max-width: 100%">
+				${getElementTemplate()}
+				${file.queueItem?.status === 'waiting' ? html`<umb-temporary-file-badge></umb-temporary-file-badge>` : nothing}
+			</div>
+		`;
+
+		function getElementTemplate() {
+			switch (type) {
+				case 'audio':
+					return html`<umb-input-upload-field-audio .path=${file.path}></umb-input-upload-field-audio>`;
+				case 'video':
+					return html`<umb-input-upload-field-video .path=${file.path}></umb-input-upload-field-video>`;
+				case 'image':
+					return html`<umb-input-upload-field-image .path=${file.path}></umb-input-upload-field-image>`;
+				case 'svg':
+					return html`<umb-input-upload-field-svg .path=${file.path}></umb-input-upload-field-svg>`;
+				case 'file':
+					return html`<umb-input-upload-field-file
+						.path=${file.path}
+						.file=${file.file as any}></umb-input-upload-field-file>`;
+			}
+		}
+	}
+
+	#getFileTypeFromPath(path: string): 'audio' | 'video' | 'image' | 'svg' | 'file' {
+		// Extract the MIME type from the data URL
+		if (path.startsWith('data:')) {
+			const mimeType = path.substring(5, path.indexOf(';'));
+			if (mimeType === 'image/svg+xml') return 'svg';
+			if (mimeType.startsWith('image/')) return 'image';
+			if (mimeType.startsWith('audio/')) return 'audio';
+			if (mimeType.startsWith('video/')) return 'video';
+		}
+
+		// Extract the file extension from the path
+		const extension = path.split('.').pop()?.toLowerCase();
+		console.log('extension', extension, path);
+		if (!extension) return 'file';
+		if (['svg'].includes(extension)) return 'svg';
+		if (['mp3', 'weba', 'oga', 'opus'].includes(extension)) return 'audio';
+		if (['mp4', 'mov', 'webm', 'ogv'].includes(extension)) return 'video';
+		if (['jpg', 'jpeg', 'png', 'gif'].includes(extension)) return 'image';
+
+		return 'file';
 	}
 
 	#renderButtonRemove() {
-		if (!this._currentFiles.length && !this._filePaths.length) return;
+		if (!this._files.length && !this._files.length) return;
```

Comment:
It looks like you are checking for the same thing twice - consider changing it to something like this:

```suggestion
		if (!this._files.length) return;
```